v1/topdown/json.go

```diff
@@ -103,7 +103,12 @@ func jsonRemove(a *ast.Term, b *ast.Term) (*ast.Term, error) {
 			v := aValue.Elem(i)
 			// recurse and add the diff of sub objects as needed
 			// Note: Keys in b will be strings for the index, eg path /a/1/b => {"a": {"1": {"b": null}}}
-			diffValue, err := jsonRemove(v, bObj.Get(ast.StringTerm(strconv.Itoa(i))))
+			key := ast.InternedIntegerString(i)
+			if key == nil {
+				key = ast.StringTerm(strconv.Itoa(i))
+			}
```

Comment:
We do this a lot. Would it be possible to add an optional default value to `ast.InternedIntegerString()` that is wrapped in a term and returned if an interned term can't be found?

Reply:
I only find 2 places where we do this. Rule of 3 :) It's a good idea though, so I'll consider it for the next phase. Just trying to keep the changeset down a bit here.

Reply:
This got done with the rewrite now.